### PR TITLE
Logout page

### PR DESCRIPTION
**Pull request by shrey27**
Added Features
- Sign out page, a landing page after user has logged out
- Remember me functionality added to both sign-up and sign-in pages
- See password while typing feature added on both signin and signup pages
- error handling for 404 and 500 cases
- code refactor based on chosen authentication method

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,7 +1,27 @@
 import { availableRoutes } from './routes';
 
+import { useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const ScrollToTop = () => {
```
Comment (abhisheksunil2201) [inline src/App.js:6]: You could extract scroll to top into a different file maybe into a folder named utils. Just a suggestion

**Review by abhisheksunil2201 — APPROVED** [review]
I feel you could use better naming of variables in a few places. everything else seems good.

File: src/App.js (hunk 1 above)
Comment (shrey27) [reply src/App.js:6]: yeah i will do that

File: src/frontend/products/index.js [hunk 2]
```diff
@@ -318,7 +329,7 @@ export default function Products() {
               {/* {wishListError && (
```
Comment (varnita619) [inline src/frontend/products/index.js:329]: remove unwanted comments

**Review by varnita619 — APPROVED** [review]
folder structure is well arranged 
every possible code is extracted out

**Pull request merged**